### PR TITLE
feat/savax-vault and updating tests

### PR DESCRIPTION
**Pull request by fqlx**
- feat/savax-vault Adding staked avax vault
- Removed `BLOCK_NUMBER_NEW` and updated `BLOCK_NUMBER`
- Updated the test cases to latest addresses
- Update addresses to the latest blocks (except for the yearn tests - need to investigate more)
- Use ManualVolOracle in RibbonThetaVault tests
- Hardhat gas reporter
- Note: we need a separate MR for sAVAX pricer

### CODE REVIEW

File: constants/constants.ts [hunk 1]
```diff
@@ -104,6 +104,8 @@ export const WSTETH_ADDRESS = {
 };
 
 export const LDO_ADDRESS = "0x5A98FcBEA516Cf06857215779Fd812CA3beF1B32";
+export const BENQI_ADDRESS = "0x8729438EB15e2C8B576fCc6AeCdA6A148776C0F5";
+export const SAVAX_AVAX_TJ_POOL = "0x4b946c91C2B1a7d7C40FB3C130CdfBaf8389094d";
```
Comment (kenchangh) [inline constants/constants.ts:108]: I don't see these addresses being used, can we remove them?

File: test/RibbonThetaVault.ts [hunk 2]
```diff
@@ -437,18 +472,20 @@ function behavesLikeRibbonOptionsVault(params: {
...
-          ? ETH_USDC_POOL[chainId]
-          : WBTC_USDC_POOL[chainId]
-      );
+      await volOracle.setAnnualizedVol(ETH_USDC_POOL[chainId], 106480000);
+      await volOracle.setAnnualizedVol(WBTC_USDC_POOL[chainId], 106480000);
+
+      for (let i = 0; i < 4; i++) {
+        const topOfPeriod = (await time.getTopOfPeriod()) + time.PERIOD;
+        await time.increaseTo(topOfPeriod);
+      }
```
Comment (kenchangh) [inline test/RibbonThetaVault.ts:488]: Trying to understand this... why do we have to increase the time to top of period 4 times?

Comment (fqlx) [reply test/RibbonThetaVault.ts]: Fixed, I had to increase the option's expiry date to the future week.

File: constants/constants.ts (hunk 1 above)
Comment (fqlx) [reply constants/constants.ts]: Removed.

**Review by kenchangh — APPROVED** [review]
LGTM

**Pull request merged**